File: aula#7/pizza/PizzaCamarao.py
from abc import ABCMeta, abstractmethod
from Pizza import Pizza

# ProdutoConcreto
class PizzaCamarao(Pizza):

    def servir(self, PizzaVeg):
        print(f'{type(self).__name__} é servida com camarão na {type(PizzaVeg).__name__}')

File: aula#7/AbstractFactory_java/Pizza/PizzaCamarao.java
package Pizza;

//ConcreteProductB1
class PizzaCamarao implements Pizza {

 @Override
 public void servir(PizzaVeg pizzaVeg) {
     System.out.println(getClass().getSimpleName() + " é servida com camarão na " + pizzaVeg.getClass().getSimpleName());
 }
}